packages/blocks/google-cloud/src/lib/actions/get-recommendations-cli-action.ts

```diff
@@ -0,0 +1,298 @@
+import { createAction, Property } from '@openops/blocks-framework';
+import {
+  getUseHostSessionProperty,
+  googleCloudAuth,
+  handleCliError,
+  tryParseJson,
+} from '@openops/common';
+import { runCommand, runCommands } from '../google-cloud-cli';
+
+export const getRecommendationsAction = createAction({
+  auth: googleCloudAuth,
+  name: 'google_get_recommendations_cli',
+  description: 'Fetch the recommendations for the selected recommenders',
+  displayName: 'Get Recommendations',
+  props: {
+    useHostSession: getUseHostSessionProperty(
+      'Google Cloud',
+      'gcloud auth login',
+    ),
+    filterBySelection: Property.Dropdown<string>({
+      displayName: 'Choose filter',
+      description:
+        'Select whether to filter by billing account, folder ID, organization ID, or project ID.',
+      required: true,
+      refreshers: [],
+      options: async () => {
+        return {
+          disabled: false,
+          options: [
+            {
+              label: 'Filter by Billing Account',
+              value: 'billingAccount',
+            },
+            {
+              label: 'Filter by Organization ID',
+              value: 'organization',
+            },
+            {
+              label: 'Filter by Project ID',
+              value: 'project',
+            },
+            {
+              label: 'Filter by Folder ID',
+              value: 'folder',
+            },
+          ],
+        };
+      },
+    }),
+    filterByProperty: Property.DynamicProperties({
+      displayName: '',
+      required: true,
+      refreshers: [
+        'auth',
+        'useHostSession',
+        'useHostSession.useHostSessionCheckbox',
+        'filterBySelection',
+      ],
+      props: async ({
+        auth,
+        useHostSession,
+        filterBySelection,
+      }): Promise<{ [key: string]: any }> => {
+        const shouldUseHostCredentials =
+          (useHostSession as { useHostSessionCheckbox?: boolean })
+            ?.useHostSessionCheckbox === true;
+
+        if (!auth && !shouldUseHostCredentials) {
+          return {
+            markdown: Property.MarkDown({
+              value: 'Please authenticate to see the filter options.',
+            }),
+          };
+        }
+
+        if (!filterBySelection) {
+          return {
+            markdown: Property.MarkDown({
+              value:
+                'Please select a filter from the dropdown to see the options.',
+            }),
+          };
+        }
+
+        return await getScopeOptionProperty(
+          auth,
+          shouldUseHostCredentials,
+          filterBySelection,
+        );
+      },
+    }),
+    recommenders: getRecommendersDropdown(),
+    location: Property.ShortText({
```

Comment:
@bigfluffycookie is this like regions? is there a static list?

Reply:
a